src/H5Gtraverse.c

```diff
@@ -666,8 +669,14 @@ H5G__traverse_real(const H5G_loc_t *_loc, const char *name, unsigned target, H5G
                     pline = &def_pline;
 
                 /* Create the intermediate group */
-                /* XXX: Should we allow user to control the group creation params here? -QAK */
-                gcrt_info.gcpl_id    = H5P_GROUP_CREATE_DEFAULT;
+                gcrt_info.gcpl_id = H5P_GROUP_CREATE_DEFAULT;
+                if ((ocrt_info = ((H5L_trav_cr_t *)op_data)->ocrt_info) != NULL) {
+                    if (ocrt_info->obj_type == H5O_TYPE_GROUP)
+                        gcrt_info.gcpl_id = ((H5G_obj_create_t *)ocrt_info->crt_info)->gcpl_id;
+                    else if (ocrt_info->obj_type == H5O_TYPE_DATASET)
+                        gcrt_info.gcpl_id = ((H5D_obj_create_t *)ocrt_info->crt_info)->dcpl_id;
```

Comment:
This seems like a poor reason to include the H5Dpkg header.  Please add a private 'getter' routine to the H5D interface instead.

Reply:
Actually, it looks like there should be 3 new private routines: a 'getter' in the H5L interface (so the H5L_trav_cr_t struct stays hidden from other packages), and a 'getter' for the H5G and H5D interfaces that retrieve the GCPL & DCPL IDs, respectively.

Reply:
I implemented the 3 getters for H5L, H5G, and H5D interfaces.
However, I got the following CI build errors: 
 /usr/bin/ld: ../src/.libs/libhdf5.so: undefined reference to `H5D_OBJ_ID'
collect2: error: ld returned 1 exit status

I couldn't figure out what went wrong for the macro H5D_OBJ_ID in the H5D interface.  Anyone knows what might be the problem?

Reply:
Not certain about this, since H5Gtraverse included H5Dprivate.h. Maybe delete your build directory and re-run autogen.sh?

Reply:
The failures are from the CI actions. Not from my own build on jelly.